type-hinting/README.md

```diff
@@ -0,0 +1,169 @@
+# Type Hinting
+
+Author: [@FrostyX](https://github.com/frostyx)
+
+This is not a case against type hinting (static types), this is a case
+against over-using them.
+
+## Good motivations
+
+- Type hints nicely complement docstrings. Docstrings are great for
+  describing the purpose of a function (or class, file) but they suck
+  for describing their arguments and return value types because they
+  can be untrue. Type hints nicely complement this.
+- Type hints can reveal some errors
+
+
+## Bad motivations
+
+### A bulletproof solution
+
+Static types are not a bulletproof solution for revealing type-related
+bugs. IIRC some study found that ~80% of type-related errors are
```

Comment:
I found [darker](https://github.com/akaihola/darker) project which should be able to check the code incrementally - it should only format and check these parts which are currently changed in git commits. See https://dev.to/akaihola/improving-python-code-incrementally-3f7a

It should contains tool such tools as [black](https://github.com/psf/black), [isort](https://pycqa.github.io/isort/), pylint, flake8, mypy and codecov
Maybe it would be worth to investigate what this tool can do and if we want to use some of the tools it offers.